nilearn/plotting/html_stat_map.py

```diff
@@ -1,157 +1,496 @@
+        heuristics are applied based upon the background image intensity.
+        Accepted float values, where a typical scan is between -2 and 2
+        (-2 = increase constrast; 2 = decrease contrast), but larger values
+        can be used for a more pronounced effect. 0 means no dimming.
+    vmax : float, or None (default=None)
+        max value for mapping colors.
+        If vmax is None and symmetric_cmap is True, vmax is the max
+        absolute value of the volume.
+        If vmax is None and symmetric_cmap is False, vmax is the max
+        value of the volume.
+    resampling_interpolation : string, optional (default nearest)
+        The interpolation method for resampling
+        See nilearn.image.resample_img
+    opacity : float in [0,1] (default 1)
+        The level of opacity of the overlay (0: transparent, 1: opaque)
 
     Returns
     -------
-    StatMapView : plot of the stat map.
-        It can be saved as an html page or rendered (transparently) by the
-        Jupyter notebook.
-
-    """
-    change_warn_msg = ('view_stat_map() is under actve development. \n'
-                       'Its interface and functionality may change '
-                       'in future without warning. \nPlease do not rely on it '
-                       'in critical and production environments.'
-                       )
-    warnings.warn(message=change_warn_msg, category=FutureWarning)
-    
-    stat_map_img = check_niimg_3d(stat_map_img, dtype='auto')
-    if bg_img == 'MNI152':
-        bg_img = datasets.load_mni152_template()
-    if bg_img is not None and bg_img is not False:
-        bg_img = image.load_img(bg_img)
-        stat_map_img = image.resample_to_img(stat_map_img, bg_img)
-        bg_mask = image.new_img_like(bg_img, bg_img.get_data() != 0)
-        stat_map_img = image.new_img_like(
-            stat_map_img, stat_map_img.get_data() * bg_mask.get_data())
-        encoded_bg = '"{}"'.format(_encode_nii(bg_img))
-    else:
-        encoded_bg = 'null'
-    colors = colorscale(
-        cmap, stat_map_img.get_data().ravel(),
-        threshold=threshold, symmetric_cmap=symmetric_cmap, vmax=vmax)
-    abs_threshold = colors['abs_threshold']
-    abs_threshold = 0. if abs_threshold is None else float(abs_threshold)
-    papaya_cmaps = _to_papaya(
-        colors['cmap'], colors['norm'],
-        abs_threshold, colors['vmax'], colors['symmetric_cmap'])
-    stat_map_params = {
-        'min': abs_threshold, 'max': float(colors['vmax']),
-        'cmap': papaya_cmaps, 'symmetric': colors['symmetric_cmap']}
-    html = get_html_template('stat_map_template.html')
-    html = html.replace(
-        'INSERT_STAT_MAP_DATA_HERE', '"{}"'.format(_encode_nii(stat_map_img)))
-    html = html.replace('INSERT_MNI_DATA_HERE', encoded_bg)
-    html = html.replace('INSERT_STAT_MAP_PARAMS_HERE',
-                        json.dumps(stat_map_params))
-    return StatMapView(html)
+    html : the html viewer object.
```

Comment:
The variable you are returning is not `html`, it is `view`, so you should write that.
In fact, rename the variable in the code & docstring to `html_view` .

Reply:
ok, that works.

Reply:
actually will go with view_html. There is also a view_json, which gets converted into view_html. Before view_json was called sprite, because it's got the sprite data in it, but that is confusing with the sprite data itself (also called sprite in the code...), and view_json also includes the colormap, the background and the parameters.

Reply:
I think this is working: view_json comes with two functions, view_json_data (to populate the object with base64 data), and view_json_params (to populate the object with all the viewer parameters).

Reply:
and html_brainsprite gets renamed _view_json_to_html. It all makes sense.

Reply:
`view_html` reads like a verb/function, while `html_view` is unambiguously a noun, so a variable/quantity. Along those lines, I think, `data_json_view`, `params_json_view` are unambiguous about what you are accomplishing here. 
What say you?

Reply:
second though: view_dict instead of view_json. It really is a python dictionary, that gets dumped into json only when it is inserted in html.

Reply:
No, inserting datatype into variable name is a bad practice. `view_dict` does not indicate the purpose of the variable, `view_json` does. `json_view` would be better.

Reply:
ok will go with that.

Reply:
I did use _json_view_data, _json_view_params etc.

It's also a noun, and it indicates that all those functions are dealing with separate aspects of json_view.